### PR TITLE
[Northumberland] Assign user to report based on updates from Alloy

### PR DESCRIPTION
**Pull request by nephila-nacrea**
Strictly speaking this code is cobrand-agnostic; the Northumberland-specific code is handled in open311-adapter.

For https://github.com/mysociety/societyworks/issues/3978.

[skip changelog]

### CODE REVIEW

File: bin/fetch [hunk 1]
```diff
@@ -22,6 +22,11 @@ use Getopt::Long::Descriptive;
 use Open311::GetServiceRequests;
 use Open311::GetServiceRequestUpdates;
```
Comment (nephila-nacrea) [inline bin/fetch:24]: Not sure if this is needed, but the script was failing on 'Can't call method "get" on an undefined value at /Users/victoria/work/my-society/fixmystreet-local/commonlib/perllib/mySociety/Locale.pm line 22.' after adding my changes 🤔

Comment (dracos) [reply bin/fetch:24]: Can't hurt! You're saying it didn't do it without your changes? Not sure why you would get this - it means that translations haven't been set up and it's trying to translate something, but nothing you've changed should involve a translation?

Comment (dracos) [reply bin/fetch:24]: Was it actually fetching an update when giving this error?

File: perllib/Open311/UpdatesBase.pm [hunk 2]
```diff
@@ -188,6 +188,36 @@ sub _process_update {
...
     }
 
+    # Assign admin user to report if 'assigned_user_*' fields supplied
+    if ( $request->{extras} && $request->{extras}{assigned_user_email} ) {
+        my $assigned_user_email = $request->{extras}{assigned_user_email};
+        my $assigned_user_name  = $request->{extras}{assigned_user_name};
+
+        my $assigned_user
+            = FixMyStreet::DB->resultset('User')
+            ->search( { email => $assigned_user_email } )->first;
```
Comment (dracos) [inline perllib/Open311/UpdatesBase.pm:198]: Could this use `find()`? That is set up to make sure it only searches verified email addresses.

File: bin/fetch (hunk 1 above)
Comment (nephila-nacrea) [reply bin/fetch:24]: Hmm, I'll have another look at this - will try to get the other fixes for this PR done first :-)

**Review by dracos — APPROVED** [review]
Thanks :)

**Pull request merged**